src/dotnet/commands/dotnet-build/ProjectGraphCollector.cs

```diff
@@ -43,7 +43,7 @@ private ProjectGraphNode TraverseProject(ProjectDescription project, IDictionary
                 foreach (var dependency in project.Dependencies)
                 {
                     LibraryDescription libraryDescription;
-                    if (lookup.TryGetValue(dependency.Name, out libraryDescription))
+                    if ((lookup.TryGetValue(dependency.Name, out libraryDescription)) && (!libraryDescription.Identity.Name.Equals(project.Identity.Name)))
```

Comment:
Ends up that the nuget restore will still think it is a cyclical reference if the dependency has the same name, regardless of the type. Due to this, I don't think we need to worry about only failing for project types since the same exact thing happens regardless of the type